src/patterns/dosdp-patterns/occupational_exposure_to_material.yaml

```
@@ -0,0 +1,74 @@
+---
+pattern_name: occupational_exposure_to_material
+pattern_iri: http://purl.obolibrary.org/obo/ecto/patterns/occupational_exposure_to_material.yaml
+
+description: 'Use this pattern to create compound term that categorise
+    exposures to various materials from the environment related to some
+    occupations.'
+
+# examples
+# occupational fume exposure
+# occupational grain dust exposure
+# occupational exposure to urea-based herbicide
+
+contributors:
+  - https://orcid.org/0000-0001-8314-2140  # Ray Stefancsik
+
+classes:
+  occupation: NCIT:C25193
+  exposure_event: ExO:0000002
+  material: BFO:0000040  # material entity
+
+relations:
+  has_exposure_stimulus: RO:0002309
+
+annotationProperties:
+  exact_synonym: oio:hasExactSynonym
+  related_synonym: oio:hasRelatedSynonym
+  xref: oio:hasDbXref
+  contributor: dct:contributor
+
+vars:
+  material: "'material'"
+
+data_list_vars:
+  orcid: xsd:anyURI  # ORCID of contributor
+  exact_synonyms: xsd:string  # add a list of exact synonyms
+  xrefs: xsd:anyURI  # add a list of external references
+
+name:
+  text: "occupational %s exposure"
+  vars:
+    - material
+
+def:
+  text: "Exposure to %s which is related to some occupation(s)."
+  vars:
+    - material
+  annotations:
+    - annotationProperty: xref
+      text: "AUTO:patterns/patterns/occupational_exposure_to_material"
+
+annotations:
+  - annotationProperty: exact_synonym
+    text: "occupational exposure to %s"
+    vars:
+      - material
+    annotations:
+      - annotationProperty: xref
+        text: "AUTO:patterns/patterns/proportion_of_attributes"
+
+  - annotationProperty: contributor
+    value: orcid
+
+exact_synonym:
+  value: exact_synonyms
+
+xref:
+  value: xrefs
+
+equivalentTo:
+  text: "'exposure_event'
+        and ('has_exposure_stimulus' some ('occupation' and %s))"
```

```suggestion
        and ('has_exposure_stimulus' some 'occupation')
        and ('has_exposure_stimulus' some %s)
        "
```
This suggestion is based on feedback on another EQ here: https://github.com/obophenotype/bio-attribute-ontology/pull/312#issuecomment-1910897281

---

The previous intersection is definitely wrong and will lead to unsats

Separation is better but I don't think occupation is a stimulus. @diatomsRcool how does an exposure scientist think about this? Do we need a context relation?

Or we could simply have a more specific genus of "occupational exposure"

---

> The previous intersection is definitely wrong and will lead to unsats
> 

Sorry, but I am not familiar with the distributive properties of these axioms. Before I uploaded the DOSDP pattern I had checked in Protege (ELK 0.5.0) with the three examples in `occupational_exposure_to_material.tsv`, but I did not find any unsats. However, I accept that my EQ still could be wrong and mean something different from what I intended:) ...

> Separation is better but I don't think occupation is a stimulus. @diatomsRcool how does an exposure scientist think about this? Do we need a context relation?

FYI: My inspiration for using 'occupation' as a stimulus is coming from https://github.com/EnvironmentOntology/environmental-exposure-ontology/blob/master/src/patterns/dosdp-patterns/exposure_to_occupation.yaml

---

Occupational exposures have been a thorn in my side for a bit. I guess now there is no ignoring any longer. Chris is right. Occupations aren't really exposures. More like contexts for exposures. An exposure to lead could be part of an occupation or something like that. I'm not sure what the correct relation would be. Is there a context relation?

The exposure to occupation pattern that is currently in ECTO was developed so that we could compose complex exposures for things like working in a battery factory which would be combinations of exposures to many different things. Lauren's use case needed that. Let's make a different pattern for this.

---

Would it be crazy to do:

```
equivalentTo:
  text: "'occupational_exposure_event'
        and ('has_exposure_stimulus' some %s)"
```

or even simpler:

```
equivalentTo:
  text: "'occupational_exposure_event'
        and %s"
```

So you can say something like `"occupational exposure event" and "fume exposure"`, or, if there are use cases for the first option, `"occupational exposure event" and has_exposure_stimulus some "fume"`?

---

Yep, this is what I meant by 

> Or we could simply have a more specific genus of "occupational exposure"

Regardless of how it's axiomatized there is the lopsided xmas tree antipattern here. "occupational X exposure" will be a child of "X exposure" as expected. But do we need to make non-occupational children? Someone annotating to an unadorned "X exposure" may have an implicit assumption of directness.

A slightly radically different way of doing this is to interpret the root ecto term as being direct, and axiomatized occupational exposure as "occupational exposure and has-part some <direct exposure term>". This gives us more fine grained control in weeding out occupational exposures in queries but at the expense of making it harder to do the inclusive query

---

The use cases I've been working with so far would need a priority placed on being able to query an exposure to lead across all different types of mixed exposures rather than querying occupational exposures specifically

---

Is an occupational exposure not a direct exposure? 

I am pretty sure we are overthinking this. Maybe its best to conceptualise ECTO as a vocabulary with a sane hierarchy (where occupational exposure classifies under exposure, and exposure has no implicit assumption of "directness"). We should probably not even export the logical definitions in the main release product; just the relaxed version with the stimuli and perhaps the route. 

For now, I am not convinved that `But do we need to make non-occupational children` is enough of a concern to avoid the "more specific genus of "occupational exposure"" solution...